From ec67e2cae892ac278083e5dc14271e0a6bb2383a Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Tue, 26 Apr 2022 14:54:39 +0200
Subject: [PATCH 168/841] feat(fdts): add watchdog support for STM32MP2

This patch add iwdg1 watchdog support on STM32MP2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2680a2ed94ecf35ba0dff37333bb650fbcb760c0 feat(fdts): add watchdog support for STM32MP2]
Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I16c25f6b8a1f10b591c8248cae881b575d04d1da
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/249555
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabien DESSENNE <fabien.dessenne@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp251.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 943441329..43a92393e 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -296,6 +296,14 @@
 			};
 		};
 
+		iwdg1: watchdog@44010000 {
+			compatible = "st,stm32mp2-iwdg";
+			reg = <0x44010000 0x400>;
+			clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
+			clock-names = "pclk", "lsi";
+			status = "disabled";
+		};
+
 		rcc: rcc@44200000 {
 			compatible = "st,stm32mp25-rcc";
 			reg = <0x44200000 0x10000>;
-- 
2.47.2

